### PR TITLE
issue-191 Incorrect module_path on Windows

### PR DESCRIPTION
**Pull request by Stychnevsky**
https://github.com/reportportal/agent-python-pytest/issues/191
Fix bag with incorrect module path on Windows OS

### CODE REVIEW

File: pytest_reportportal/service.py [hunk 1]
```diff
@@ -529,14 +529,18 @@ def _add_item_hier_parts_other(item_parts, item, item_type, hier_flag,
...
-            if type(part) is item_type:
+            if isinstance(part, item_type):
 
                 if item_type is Module:
                     module_path = str(
                         item.fspath.new(dirname=rp_name,
                                         basename=part.fspath.basename,
                                         drive=""))
-                    rp_name = module_path if rp_name else module_path[1:]
+                    if module_path.startswith(':\\', 1):
```
Comment (iivanou) [inline pytest_reportportal/service.py:539]: It's better to check the operation system as well:

```
# mkdir C:\\
# ls
C:\
```

File: pytest_reportportal/service.py [hunk 2]
```diff
@@ -529,14 +529,18 @@ def _add_item_hier_parts_other(item_parts, item, item_type, hier_flag,
...
+                        stripped_module_path = module_path[3:]
```
Comment (iivanou) [inline pytest_reportportal/service.py:540]: Can you please try not to use slicing? Search through the os.path for better solutions.

File: pytest_reportportal/service.py (hunk 1 above)
Comment (iivanou) [inline pytest_reportportal/service.py:539]:
```
stripped_module_path = something
if windows:
    stripped_module_path = anything
```

File: pytest_reportportal/service.py [hunk 3]
```diff
@@ -529,14 +529,18 @@ def _add_item_hier_parts_other(item_parts, item, item_type, hier_flag,
...
+                    else:
+                        stripped_module_path = module_path[1:]
+                    rp_name = module_path if rp_name else stripped_module_path
```
Comment (iivanou) [inline pytest_reportportal/service.py:543]: rp_name = rp_name or stripped_module_path

File: tests/test_service.py [hunk 4]
```diff
@@ -84,7 +86,30 @@ def __iter__(self):
     rp_service.collect_tests(mocked_session)
     rp_service.start_pytest_item(mocked_item)
 
-    expect(mocked_item_start.call_count == 1, 'One HTTP POST sent')
+    expect(mocked_item_start.call_count == 2, 'Two HTTP POST sent')
```
Comment (iivanou) [inline tests/test_service.py:89]: Why did u change it?

File: tests/test_service.py [hunk 5]
```diff
@@ -84,7 +86,30 @@ def __iter__(self):
...
+
+@pytest.mark.parametrize('os_type', ('windows', 'unix'))
+def test_adding_item_to_hierarchy(mocked_item, os_type):
+    """ Test adding path of module to hierarchy on both OS
+
+    :param mocked_item:       a mocked test item
+    :param os_type:           type of operation system
+    """
+    parent_path = mocked_item.parent.fspath
+    if os_type == 'windows':
```
Comment (iivanou) [inline tests/test_service.py:103]: Why did not u use paths directly in parametrization instated of checking for os_type?

File: tests/test_service.py [hunk 6]
```diff
@@ -84,7 +86,30 @@ def __iter__(self):
...
+    else:
+        parent_path = parent_path.new(dirname='/path/to/item')
+    rp_name = PyTestServiceClass._add_item_hier_parts_other(
+        item_parts=PyTestServiceClass._get_item_parts(mocked_item),
+        item=mocked_item,
+        item_type=Module,
+        hier_flag=False,
+        report_parts=[],
+        rp_name="")
+    expect(rp_name == 'module.py')
```
Comment (iivanou) [inline tests/test_service.py:114]: Soft asserts should be used only if u have multiple checks within one test case.

File: pytest_reportportal/service.py (hunk 3 above)
Comment (Stychnevsky) [reply pytest_reportportal/service.py:543]: ok, if we don't need to change rp_name when it already exist, seems that we don't need module_path variable and there are no problems with OS type.

File: tests/test_service.py (hunk 6 above)
Comment (Stychnevsky) [reply tests/test_service.py]: fixed

File: tests/test_service.py (hunk 5 above)
Comment (Stychnevsky) [reply tests/test_service.py]: fixed

File: tests/test_service.py (hunk 4 above)
Comment (Stychnevsky) [reply tests/test_service.py]: When I mock mocked_module type (mock.create_autospec(Module)), call count changed to 2.
So, now I return call_count to 1, and create separated fixtures to fix it

**Pull request closed without merging**